repair_order_template/models/repair_order_template_line.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Camptocamp SA (https://www.camptocamp.com).
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class RepairOrderTemplateLine(models.Model):
+    _name = "repair.order.template.line"
+    _description = "Repair Order Template Line"
+    _check_company_auto = True
+
+    template_id = fields.Many2one(
+        "repair.order.template",
+        required=True,
+        readonly=True,
+        ondelete="cascade",
+    )
+    company_id = fields.Many2one(
+        related="template_id.company_id",
+        store=True,
+    )
+    type = fields.Selection(
+        selection=lambda self: (
+            self.env["stock.move"]
+            ._fields["repair_line_type"]
+            ._description_selection(self.env)
+        ),
+        required=True,
+    )
+    product_id = fields.Many2one(
+        "product.product",
+        string="Product",
+        domain="""
+            [
+                ('type', 'in', ['product', 'consu']),
+                ('company_id', 'in', [False, company_id])
+            ]
+        """,
+        check_company=True,
+        required=True,
+    )
+    product_uom_category_id = fields.Many2one(
+        related="product_id.uom_id.category_id",
+    )
+    product_uom = fields.Many2one(
+        "uom.uom",
+        string="Unit of Measure",
+        domain="[('category_id', '=', product_uom_category_id)]",
```

Comment:
we use here computed instead of related to add a domain?

Reply:
Hey @yankinmax !

Sorry, I didn't understand the question.

The domain here is to ensure the user selects only UOM's that have the same category as the product's uom. Notice the field is computed, but has readonly=False.

Reply:
Why not using the related attribute on the field: `related="product_id.uom_id"

Reply:
That's not the same thing.

Upon write, it would update the product's `uom_id` instead.
Here, we essentially want a default value that comes from the product